pkg/victoriametrics/source.go

```diff
@@ -330,23 +398,27 @@ func (s Source) HasMetrics(start, end time.Time) (bool, error) {
 		Str("url", url).
 		Msg("Sending GET query request to Victoria Metrics endpoint")
 
-	req := fasthttp.AcquireRequest()
-	defer fasthttp.ReleaseRequest(req)
-
 	req.Header.SetMethod(fasthttp.MethodGet)
 	req.SetRequestURI(url)
 	req.Header.Set(fasthttp.HeaderAcceptEncoding, "gzip")
 
 	resp, err := s.c.DoWithTimeout(req, requestTimeout)
 	defer fasthttp.ReleaseResponse(resp)
+
 	if err != nil {
 		return false, errors.Wrap(err, "failed to send HTTP request to victoria metrics")
 	}
 
 	body := gzipDecode(copyBytesArr(resp.Body()))
+
 	if status := resp.StatusCode(); status != fasthttp.StatusOK {
-		return false, errors.Errorf("non-OK response from victoria metrics: %d: %s", status, body)
+		if strings.Contains(body, "cannot select more than -search.maxSamplesPerQuery") {
+			return s.splitContainsMetrics(start, end, body)
+		} else {
+			return false, errors.Errorf("non-OK response from victoria metrics: %d: %s", status, body)
+		}
```

Comment:
`else` redundant here

```suggestion
		if strings.Contains(body, "cannot select more than -search.maxSamplesPerQuery") {
			return s.splitContainsMetrics(start, end, body)
		}
		
		return false, errors.Errorf("non-OK response from victoria metrics: %d: %s", status, body)
		
```